.travis.yml

```diff
@@ -0,0 +1,98 @@
+  # $TRAVIS_BRANCH documentation - https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+  - if [[ "$TRAVIS_BRANCH" =~ ^[1-9]$ ]] || [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}.x-dev"; elif [[ "$TRAVIS_BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${TRAVIS_BRANCH}"; else export COMPOSER_ROOT_VERSION="dev-${TRAVIS_BRANCH}"; fi
+
+  # PHPENV
+  - phpenv rehash
+  - phpenv config-rm xdebug.ini || true
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then memlimit='8192M'; else memlimit='4096M'; fi
+  - echo "memory_limit = ${memlimit}" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - echo 'always_populate_raw_post_data = -1' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+
+  # COMPOSER
+  - if [ $COMPOSER_VERSION ]; then composer self-update "$COMPOSER_VERSION"; elif [ $(php -r 'echo (int) version_compare(phpversion(), "7.3.0", "<=");') = "1" ]; then composer self-update 1.9.3; else composer self-update; fi
+  - composer --version
+  # Require sminne phpunit in .travis.yml rather than require-dev in composer.json because they're not compatible with behat-extension:^1 which uses phpunit 3
+  - composer require --no-update sminnee/phpunit:^5.7 sminnee/phpunit-mock-objects:^3.4.5
+  - if [[ $(composer validate 2>&1 | tr '\n' ' ') =~ ' is valid' ]]; then cat composer.json; fi
+  - composer update --prefer-source --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile $COMPOSER_INSTALL_ARG
+  - composer show
+
+  # silverstripe-installer - includes files required for minimal site setup e.g. Page, PageController
+  - composer create-project silverstripe/installer:^3 ~/myinstaller
+  - cp -R ~/myinstaller/assets .
+  - cp -R ~/myinstaller/mysite .
+  - cp -R ~/myinstaller/themes .
+  - cp ~/myinstaller/.htaccess .
+  - cp ~/myinstaller/index.php .
+
+  # turn off php 7.4 deprecation warnings
+  - echo "error_reporting(E_ALL ^ E_DEPRECATED);" >> mysite/_config.php
+
+  # add _ss_environment.php file
+  - PWD=$(pwd)
+  - echo "<?php" > _ss_environment.php
+  - echo "define('SS_ENVIRONMENT_TYPE', 'dev');" >> _ss_environment.php
+  - echo "define('SS_TRUSTED_PROXY_IPS', '*');" >> _ss_environment.php
+  - echo "define('SS_DATABASE_SERVER', '127.0.0.1');" >> _ss_environment.php
+  - echo "define('SS_DATABASE_USERNAME', 'root');" >> _ss_environment.php
+  - echo "define('SS_DATABASE_PASSWORD', '');" >> _ss_environment.php
+  - echo "define('SS_DATABASE_CHOOSE_NAME', 1);" >> _ss_environment.php
+  - echo "define('SS_DEFAULT_ADMIN_USERNAME', 'admin');" >> _ss_environment.php
+  - echo "define('SS_DEFAULT_ADMIN_PASSWORD', 'password');" >> _ss_environment.php
+  - echo "define('SS_HOST', 'localhost');" >> _ss_environment.php
+  - echo "global \$_FILE_TO_URL_MAPPING;" >> _ss_environment.php
+  - echo "\$_FILE_TO_URL_MAPPING['$PWD'] = 'http://localhost:8000';" >> _ss_environment.php
```

Comment:
Is this needed? I ran travis build on CMS3 before and didn't need all this stuff.

Reply:
Chances are it worked because the old builds were using this rather outdated helper https://github.com/silverstripe/silverstripe-travis-support